Fix bisq.properties pickup in ApiTest

alyokaz opened the pull request:
This change copies the 'bisq.properties' file in apitest/src/main/resources to each instance's DataDir on test harness startup.

This was necessary as #5718 was not copying the file into the instance's data dir but leaving it in 'build/resources/main' where it wasn't picked up by the instance.

<!-- 
- make yourself familiar with the CONTRIBUTING.md if you have not already (https://github.com/bisq-network/bisq/blob/master/CONTRIBUTING.md)
- make sure you follow our [coding style guidelines][https://github.com/bisq-network/style/issues)
- pick a descriptive title
- provide some meaningful PR description below
- create the PR
- in case you receive a "Change request" and/or a NACK, please react within 30 days. If not, we will close your PR and it can not be up for compensation.
- After addressing the change request, __please re-request a review!__ Otherwise we might miss your PR as we tend to only look at pull requests tagged with a "review required".
-->

alyokaz committed:
Fix bisq.properties pickup in ApiTest

Copy 'bisq.properties' file in apitest/src/main/resources
to each instance's DataDir on startup to allow pickup by
the instance.

alvasw commented on apitest/src/main/java/bisq/apitest/Scaffold.java:
```diff
@@ -242,28 +243,52 @@ public void installDaoSetupDirectories() {
-                    new BashCommand("rm -rf " + buildDataDir + "/bisq.properties");
-            if (rmRedundantBisqPropertiesFile.run().getExitStatus() != 0)
-                throw new IllegalStateException("Could not delete redundant bisq.properties file");
+            deleteRedundantBisqPropertiesFile();
```
I know, you didn't write this code, but do we need to delete the `bisq.properties` file? This breaks the Gradle Build Cache.

alyokaz replied:
I think it's fine to keep, its presence signals it's being used by all instances, and its absence signals explicit inclusion.

I've removed `deleteRedundantBisqPropertiesFile()` and done some refactoring on `dao-setup.gradle`, removing some redundant copying and renaming.

alvasw reviewed (changes requested):
Do we need to delete the `bisq.properties` file in the root directory? The deletion breaks the Gradle Build Cache. It's smarter to include/exclude the file explicitly [1], or even have them multiple times (for all instances) in the resources directory.

I know that you didn't write the initial file-deletion code.

[1] https://docs.gradle.org/current/dsl/org.gradle.language.jvm.tasks.ProcessResources.html

alyokaz committed:
Refactor dao-setup.gradle

Remove redundant directory moving and renaming. Make buildResourcesDir
property less brittle.

alyokaz committed:
Stop deletion of original bisq.properties during test harness setup

Stop the deletion of the original bisq.properties file after it's copied into the
instances data directories.

alvasw commented on apitest/src/main/java/bisq/apitest/Scaffold.java:
```diff
@@ -29,6 +29,7 @@
 import java.io.FileNotFoundException;
 import java.io.IOException;
 
+import java.util.Arrays;
```
This is import is not needed anymore.

alvasw commented on apitest/src/main/java/bisq/apitest/Scaffold.java:
```diff
@@ -242,28 +243,36 @@ public void installDaoSetupDirectories() {
+    private void copyBisqPropertiesFileToAppDataDir() throws IOException, InterruptedException {
+        copyBisqPropertiesFileToAppDataDir(alicedaemon);
+        copyBisqPropertiesFileToAppDataDir(bobdaemon);
+    }
```
Please reformat this to:
```diff
     }
+
     private void copyBisqPropertiesFileToAppDataDir(BisqAppConfig bisqAppConfig) throws IOException, InterruptedException {
-        if(!new File(config.baseSrcResourcesDir + "/bisq.properties").exists()) return;
+        if (!new File(config.baseSrcResourcesDir + "/bisq.properties").exists()) return;
 
         String instanceDataDir = config.rootAppDataDir + "/" + bisqAppConfig.appName;
```

alyokaz replied:
Amended as instructed.

alvasw reviewed (changes requested):
Looks good!

alvasw reviewed (approved):
utACK

alejandrogarcia83 reviewed (approved):
utACK
